web/src/compose_recipient.ts

```diff
@@ -359,10 +359,14 @@ export let update_compose_area_placeholder_text = (): void => {
     let placeholder = compose_ui.DEFAULT_COMPOSE_PLACEHOLDER;
     if (message_type === "stream") {
         const stream_id = compose_state.stream_id();
+        let topic = compose_state.topic();
+        if (empty_topic_display && !realm.realm_mandatory_topics) {
+            topic = util.get_final_topic_display_name(topic);
+        }
         placeholder = compose_ui.compute_placeholder_text({
             message_type,
             stream_id,
-            topic: compose_state.topic(),
+            topic,
```

Comment:
Hmm, would it make sense for `compose_ui.compute_placeholder_text` to be responsible for doing this translation? I think that'd be a better abstraction, but it should have no functional impact.

Reply:
Yeah.

Actually, I ended up dropping Karl's commit as it was a bit buggy -- start the compose box in CZO using "start new conversation" button or something, the placeholder shows "test general chat".

Added a commit which fixes this bug and doesn't require passing `empty_topic_display` parameter around.